pp_netlib/gt_prune.py

```diff
@@ -0,0 +1,293 @@
+    ref_order = [name for idx, name in enumerate(vertex_labels) if idx in frozenset(ref_indices)]
+    clusters_in_reference_graph = gt_print_clusters(ref_graph, ref_order, backend = backend, print_csv=False)
+    # Record the components/clusters the references are in the reference graph
+    # dict: name: ref_cluster
+    reference_clusters_in_reference_graph = {}
+    for reference_name in ref_order:
+        try:
+            reference_clusters_in_reference_graph[reference_name] = clusters_in_reference_graph[reference_name]
+        except IndexError:
+            pass
+
+    # Check if multi-reference components have been split as a validation test
+    # First iterate through clusters
+    network_update_required = False
+    for cluster_id, ref_idxs in reference_clusters_in_full_graph.items():
+        # Identify multi-reference clusters by this length
+        if len(ref_idxs) > 1:
+            check = list(ref_idxs)
+            # check if these are still in the same component in the reference graph
+            for i in range(len(check)):
+                component_i = reference_clusters_in_reference_graph[vertex_labels[check[i]]]
+                for j in range(i + 1, len(check)):
+                    # Add intermediate nodes
+                    component_j = reference_clusters_in_reference_graph[vertex_labels[check[j]]]
+                    if component_i != component_j:
+                        network_update_required = True
+                        # if backend == "GT":
+                        vertex_list, edge_list = gt.shortest_path(graph, check[i], check[j])
+                        for vertex in vertex_list:
+                            reference_vertex[vertex] = True
+                            ref_indices.add(int(vertex))
+                        # elif backend == "NX":
+                        #     print(check[i], check[j])
+                        #     vertex_list = nx.shortest_path(graph, "n"+str(check[i]), "n"+str(check[j]))
+                        #     print(vertex_list)
+                        #     for vertex in vertex_list:
+                        #         print(vertex)
+                        #         vert_idx = vertex.replace("n", "")
+                        #         print(ref_indices)
+                        #         ref_indices.add(int(vert_idx))
+                        #         print(ref_indices)
+
+
+    # update reference graph if vertices have been added
+    if network_update_required:
+        # if backend == "GT":
+        ref_graph = gt.GraphView(graph, vfilt = reference_vertex)
+        ref_graph = gt.Graph(ref_graph, prune = True) # https://stackoverflow.com/questions/30839929/graph-tool-graphview-object
+        # elif backend == "NX":
+        #     ref_graph = graph.subgraph(["n"+str(ri) for ri in ref_indices])
+
+    # Order found references as in sketch files
+    #reference_names = [vertex_labels[int(x)] for x in sorted(ref_indices)]
+    ###
```

Comment:
```suggestion
    if network_update_required:
        ref_graph = gt.GraphView(graph, vfilt = reference_vertex)
        ref_graph = gt.Graph(ref_graph, prune = True)
```

Reply:
Also, why is the final ordering no longer needed?

Reply:
the code is not using the ordered list of reference names currently, shall i remove the relevant line?

Reply:
yes, that sounds right then